src/compute_grid.cpp

```diff
@@ -0,0 +1,247 @@
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/ Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "compute_grid.h"
+
+#include "atom.h"
+#include "comm.h"
+#include "domain.h"
+#include "error.h"
+#include "force.h"
+#include "memory.h"
+#include "modify.h"
+#include "update.h"
+
+#include <cstring>
+
+using namespace LAMMPS_NS;
+
+/* ---------------------------------------------------------------------- */
+
+ComputeGrid::ComputeGrid(LAMMPS *lmp, int narg, char **arg) :
+    Compute(lmp, narg, arg), grid(nullptr), gridall(nullptr), gridlocal(nullptr)
+{
+  if (narg < 6) error->all(FLERR, "Illegal compute grid command");
+
+  array_flag = 1;
+  size_array_cols = 0;
+  size_array_rows = 0;
+  extarray = 0;
+
+  int iarg0 = 3;
+  int iarg = iarg0;
+  if (strcmp(arg[iarg], "grid") == 0) {
+    if (iarg + 4 > narg) error->all(FLERR, "Illegal compute grid command");
+    nx = utils::inumeric(FLERR, arg[iarg + 1], false, lmp);
+    ny = utils::inumeric(FLERR, arg[iarg + 2], false, lmp);
+    nz = utils::inumeric(FLERR, arg[iarg + 3], false, lmp);
+    if (nx <= 0 || ny <= 0 || nz <= 0) error->all(FLERR, "All grid dimensions must be positive");
+    iarg += 4;
+  } else
+    error->all(FLERR, "Illegal compute grid command");
+
+  nargbase = iarg - iarg0;
+
+  size_array_rows = nx * ny * nz;
+  size_array_cols_base = 3;
+  gridlocal_allocated = 0;
+}
+
+/* ---------------------------------------------------------------------- */
+
+ComputeGrid::~ComputeGrid()
+{
+  deallocate();
+}
+
+/* ---------------------------------------------------------------------- */
+
+void ComputeGrid::init() {}
```

Comment:
If this function is empty it does not need to be provided and can also be removed from the header. It is already defined in the Compute base class.

Reply:
Done